### DISCUSSION TITLE
Logic to handle cells missed by cell detection model

### DISCUSSION THREAD

**Discussion by jayeshthukarul** [deepdoctection/deepdoctection#108] [Ideas]
Can we add pipeline component which can run after cell detection service to add the cells which are missed by cell detection model? We can calculate the iou between doctr detection bounding box and already detected cells bounding box and whose iou is close to 0 we can add those bounding box as a cell.
I have coded similar logic for one of my project and it's working fine.

**Comment by JaMe76** [comment]
The approach sounds reasonable, but maybe one could generalize this refinement process such that one can generate any `LayoutType` (cells, text, title, ...) for a given word/text line that do not have any overlap. I find myself in the situation where some words did not have any overlap with text regions and therefore get lost.

There is the generic `match_anns_by_intersection`  in `mapper.match` which gives you positive mappings, so by taking the complement you can get annotations that do not match.
For the pipeline component one should have the option to operate on the whole image but also on separately some given sections (e.g. tables) so that the assignment process still works.

**Reply by jayeshthukarul** [reply]
Yes, sounds good. Actually I was about to raise this issue of lost words which is not having any overlap. Will check it how to handle this. Thanks Dr.

**Reply by JaMe76** [reply]
With respect to lost words you can try to configure the pipeline component `TextOrderService` and set `text_containers_to_text_block=True`. The reason why orphan-like words get lost is that they do not receive a reading order value. Text ordering will order higher order layout segments and in a second step words within a layout segment.  

Changing `text_containers_to_text_block` flag will treat orphan words like a text block and assign a reading order value. 

You can try that, but in some cases this causes weird behavior as words are generally much smaller layout segments and the layout ordering algorithm might not be designed for that purposes.

**Reply by jayeshthukarul** [reply]
Yes. Thanks. I tried above mentioned solution. It works but yes as you mentioned observed weird behaviour.
